packages/itwin/reports-config-widget/src/widget/components/BulkExtractor.ts

```diff
@@ -0,0 +1,190 @@
+
+export type ReportMappingType = CreateTypeFromInterface<ReportMapping>;
+
+export type ReportMappingAndMapping = ReportMappingType & {
+  mappingName: string;
+  mappingDescription: string;
+  iModelName: string;
+};
+
+export default class BulkExtractor {
+  private _reportRunIds = new Map<string, string[]>();
+  private _apiConfig: ReportsApiConfig;
+
+  constructor(apiConfig: ReportsApiConfig) {
+    this._apiConfig = apiConfig;
+  }
+
+  public async getStates(reportIds: string[]): Promise<Map<string, ExtractionStates>> {
+    const extractionClientApi = new ExtractionClient(
+      generateUrl(REPORTING_BASE_PATH, this._apiConfig.baseUrl)
+    );
+    const accessToken = await this._apiConfig.getAccessToken();
+
+    const stateByReportId = new Map<string, ExtractionStates>();
+    const stateByRunId = new Map<string, ExtractorState>();
+
+    for (const reportId of reportIds) {
+      const runs = this._reportRunIds.get(reportId);
+      if (!runs) {
+        stateByReportId.set(reportId, ExtractionStates.None);
+        continue;
+      }
+
+      const states: ExtractorState[] = [];
+      for (const runId of runs) {
+        const state = stateByRunId.get(runId);
+        if (state) {
+          states.push(state);
+        } else {
+          const runState = await this.getSingleState(runId, extractionClientApi, accessToken);
+          states.push(runState);
+          stateByRunId.set(runId, runState);
+        }
+      }
+      const finalState = this.getFinalState(states);
+      stateByReportId.set(reportId, finalState);
+    }
+    return stateByReportId;
+  }
+
+  public clearJobs() {
+    this._reportRunIds = new Map<string, string[]>();
+  }
+
+  private getFinalState(states: ExtractorState[]) {
+    const responseState = states.includes(ExtractorState.Failed) ? ExtractionStates.Failed :
+      states.includes(ExtractorState.Queued) ? ExtractionStates.Queued :
+        states.includes(ExtractorState.Running) ? ExtractionStates.Running :
+          states.includes(ExtractorState.Succeeded) ? ExtractionStates.Succeeded : ExtractionStates.Failed;
+    return responseState;
+  }
+
+  private async getSingleState(runId: string, extractionClientApi: ExtractionClient, accessToken: string): Promise<ExtractorState> {
+    try {
+      const response = await extractionClientApi.getExtractionStatus(accessToken, runId);
+      return response.state;
+    } catch (error: any) {
+      handleError(error.status);
+    }
+    return ExtractorState.Failed;
+  }
+
+  public async startJobs(reportIds: string[]) {
+    const extractionClientApi = new ExtractionClient(
+      generateUrl(REPORTING_BASE_PATH, this._apiConfig.baseUrl)
+    );
+    const accessToken = await this._apiConfig.getAccessToken();
+
+    const extractionByIModel = new Map<string, string>();
+    for (const reportId of reportIds) {
+      const IModels = (await this.fetchReportMappings(reportId, this._apiConfig)).map((m) => m.imodelId);
```

Comment:
Local variables should start with lowercase letters.

Reply:
`IModels` variable renamed to `iModels`